File: backend/services/calculator.py
"""DASP refund calculator."""
from typing import Optional

SUPER_RATE = 0.12
TAX_RATES = {
    "working_holiday": 0.65,  # 65% tax => keep 35%
    "other_temp": 0.35,        # 35% tax => keep 65%
}


def compute_refund(visa_type: str, input_mode: str, super_balance: Optional[float], gross_earnings: Optional[float]) -> dict:
    if input_mode == "balance":
        balance = float(super_balance or 0)
    else:
        balance = float(gross_earnings or 0) * SUPER_RATE
    tax_rate = TAX_RATES.get(visa_type, 0.5)
    refund = balance * (1 - tax_rate)
    return {
        "balance": round(balance, 2),
        "tax_rate": tax_rate,
        "estimated_refund": round(refund, 2),
    }

File: src/lib/server/calculator.js
// DASP refund calculator — line-for-line port of backend/services/calculator.py.

export const SUPER_RATE = 0.12;
export const TAX_RATES = {
  working_holiday: 0.65, // 65% tax => keep 35%
  other_temp: 0.35, // 35% tax => keep 65%
};

export function computeRefund(visaType, inputMode, superBalance, grossEarnings) {
  let balance;
  if (inputMode === "balance") {
    balance = Number(superBalance || 0);
  } else {
    balance = Number(grossEarnings || 0) * SUPER_RATE;
  }
  const taxRate = Object.prototype.hasOwnProperty.call(TAX_RATES, visaType) ? TAX_RATES[visaType] : 0.5;
  const refund = balance * (1 - taxRate);
  return {
    balance: Math.round(balance * 100) / 100,
    tax_rate: taxRate,
    estimated_refund: Math.round(refund * 100) / 100,
  };
}
